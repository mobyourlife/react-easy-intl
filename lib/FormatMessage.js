import React, { Component } from 'react';

class FormatMessage extends Component {
  render() {
    return <span>Hi {this.props.children}!</span>
  }
}

export default FormatMessage;
