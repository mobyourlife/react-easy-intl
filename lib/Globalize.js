export class Globalize {
  static getLocale () {
    return global.EasyIntlLocale || 'en'
  }

  static setLocale (locale) {
    global.EasyIntlLocale = locale
  }

  static getMessages () {
    if (!global.EasyIntlMessages) {
      return []
    }

    return global.EasyIntlMessages[this.getLocale()]
  }

  static getMessage (str) {
    const messages = this.getMessages()
    return messages[str] || str
  }

  static setMessages (messages) {
    global.EasyIntlMessages = messages
  }
}

export default Globalize
