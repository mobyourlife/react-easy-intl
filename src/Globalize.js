export class Globalize {
  static getLocale () {
    return window.EasyIntlLocale || 'en'
  }

  static setLocale (locale) {
    window.EasyIntlLocale = locale
  }

  static getMessages () {
    if (!window.EasyIntlMessages) {
      return []
    }

    return window.EasyIntlMessages[this.getLocale()]
  }

  static getMessage (str) {
    const messages = this.getMessages()
    return messages[str] || str
  }

  static setMessages (messages) {
    window.EasyIntlMessages = messages
  }
}

export default Globalize
