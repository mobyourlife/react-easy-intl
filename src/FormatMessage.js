import React, { Component } from 'react'
import MessageFormat from 'messageformat'
import Globalize from './Globalize'

export class FormatMessage extends Component {
  constructor (props) {
    super(props)
    this.mf = new MessageFormat(Globalize.getLocale())
  }

  render () {
    if (!this.props.children) {
      return null
    }

    const msg = Globalize.getMessage(this.props.children)
    const fn = this.mf.compile(msg)
    const text = fn(this.props)
    return <span>{text}</span>
  }
}

export default FormatMessage
