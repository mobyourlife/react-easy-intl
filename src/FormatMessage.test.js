/* global it, expect */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { render } from 'enzyme'
import Globalize from './Globalize'
import FormatMessage from './FormatMessage'

const messages = {
  en: {
    'Hi #name!': 'Hi {name}!',
    'I have #qty books.': 'I have {qty, plural, =0{no books} one{1 book} other{# books} }.'
  },
  pt: {
    'Hi #name!': 'Olá {name}!',
    'I have #qty books.': 'Eu {qty, plural, =0{não tenho nenhum livro} one{tenho 1 livro} other{tenho # livros} }.'
  }
}

beforeAll(done => {
  Globalize.setMessages(messages)
  done()
})

describe('basic tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<FormatMessage />, div)
  })

  it('renders component with text', () => {
    const component = renderer.create(
      <FormatMessage />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('#en locale tests', () => {
  beforeAll(done => {
    Globalize.setLocale('en')
    done()
  })

  it('renders passed string', () => {
    const wrapper = render(<FormatMessage name='John Doe'>Hi #name!</FormatMessage>)
    expect(wrapper.text()).toEqual('Hi John Doe!')
  })

  it('pluralizes zero', () => {
    const wrapper = render(<FormatMessage qty={0}>I have #qty books.</FormatMessage>)
    expect(wrapper.text()).toEqual('I have no books.')
  })

  it('pluralizes one', () => {
    const wrapper = render(<FormatMessage qty={1}>I have #qty books.</FormatMessage>)
    expect(wrapper.text()).toEqual('I have 1 book.')
  })

  it('pluralizes number', () => {
    const wrapper = render(<FormatMessage qty={100}>I have #qty books.</FormatMessage>)
    expect(wrapper.text()).toEqual('I have 100 books.')
  })
})

describe('#pt locale tests', () => {
  beforeAll(done => {
    Globalize.setLocale('pt')
    done()
  })

  it('renders passed string', () => {
    const wrapper = render(<FormatMessage name='John Doe'>Hi #name!</FormatMessage>)
    expect(wrapper.text()).toEqual('Olá John Doe!')
  })

  it('pluralizes zero', () => {
    const wrapper = render(<FormatMessage qty={0}>I have #qty books.</FormatMessage>)
    expect(wrapper.text()).toEqual('Eu não tenho nenhum livro.')
  })

  it('pluralizes one', () => {
    const wrapper = render(<FormatMessage qty={1}>I have #qty books.</FormatMessage>)
    expect(wrapper.text()).toEqual('Eu tenho 1 livro.')
  })

  it('pluralizes number', () => {
    const wrapper = render(<FormatMessage qty={100}>I have #qty books.</FormatMessage>)
    expect(wrapper.text()).toEqual('Eu tenho 100 livros.')
  })
})

describe('auto language detection', () => {
  it('detects without crashing', () => {
    Globalize.detectLocale()
  })
})
