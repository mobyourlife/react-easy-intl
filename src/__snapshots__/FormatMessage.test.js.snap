exports[`basic tests renders component with text 1`] = `null`;
